synapse/http/federation/srv_resolver.py

```diff
@@ -28,7 +28,7 @@
 
 logger = logging.getLogger(__name__)
 
-SERVER_CACHE = {}
+SERVER_CACHE: Dict = {}
 
 
 @attr.s(slots=True, frozen=True)
```

Comment:
Maybe make this `auto_attribs=True` while we're at it. We could do something like the following for attributes without a default:

```python
host: bytes
```

And for ones with a default:

```python
expires: int = 0
```